### PR TITLE
average time report, search employees

### LINKED ISSUE #16: Reports - Average time employees checked in/out by month
Reports - Average time employees checked in/out by month

### PR DESCRIPTION
**Pull request by AlejandroCuevasZepeda**
# Description
 - added average time report
 - added search employees
 - fixed bugs

  ## Screenshots
image
image

### CODE REVIEW

File: app/models/employee.rb [hunk 1]
```diff
@@ -19,19 +19,36 @@ class Employee < ApplicationRecord
...
+
   scope :actives, -> { where(is_active: true) }
 
+  def self.search(search_name)
+    if search_name
+      self.where('name LIKE ?', "%#{search_name}%")
+    else
+      self.order(:name)
+    end
+  end
```
Comment (vyk2rr) [inline app/models/employee.rb:35]: I suggest to add specs for this, in the same way you did for CompanyBranch model, and in the same way for methods absence_by_month, and average. That would give clarity to the code in general.

Comment (alexs) [reply app/models/employee.rb:35]: Yep we will write it. Thanks

**Review by brocha05 — APPROVED** [review]

**Review by Frcomparan — APPROVED** [review]
LGTM

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -12,4 +12,11 @@
...
- @import "bootstrap";
\ No newline at end of file
+ @import "bootstrap";
+
+ .login_button{
+    position:absolute;
+    top:0;
+    right:0;
+    margin:10px;
+ }
```
Comment (AlejandroCuriel) [inline app/assets/stylesheets/application.scss:22]: Add white space at the end of the file

File: app/models/employee.rb [hunk 3]
```diff
@@ -19,19 +19,36 @@ class Employee < ApplicationRecord
...
+
   def absence_by_month
-    attendances.where.not(check_in: nil).where('check_in BETWEEN ? AND ?', DateTime.now.beginning_of_month,
-                                               DateTime.now.end_of_month).count
+    DateTime.now.day - attendances.where(
+      'check_in BETWEEN ? AND ?',
+      DateTime.now.beginning_of_month,
+      DateTime.now.end_of_month
+    ).count
```
Comment (AlejandroCuriel) [inline app/models/employee.rb:42]: take care of the indentation of the code

**Review by AlejandroCuriel — APPROVED** [review]
Everything else looks good to me, but there are some observations to consider

**Review by juperameza — APPROVED** [review]
LLGTM

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (MrCano369x) [reply app/assets/stylesheets/application.scss:22]: ok

File: app/models/employee.rb (hunk 3 above)
Comment (MrCano369x) [reply app/models/employee.rb:42]: ok

**Pull request merged**